handling-forms/submitting-forms.md

```diff
@@ -0,0 +1,428 @@
+
+1. Pass down event handlers from `App` to the `NewStudentForm`
+1. Configure the form submission in `NewStudentForm` to lift state up
+
+## Passing Down New Student Handlers
+
+Sofia starts by making a method in `App` that adds a new student to the student data:
+
+<!-- prettier-ignore-start -->
+```js
+    const addStudentData = newStudent => {
+        // Duplicate the student list
+        const newStudentList = [...studentData];
+
+        // Logic to generate the next valid student ID
+        const nextId = Math.max(...newStudentList.map(student => student.id)) + 1;
+
+        newStudentList.push({
+            id: nextId,
+            nameData: newStudent.nameData,
+            emailData: newStudent.emailData,
+            isPresentData: false,
+        });
+
+        setStudentData(newStudentList);
+    };
+```
+<!-- prettier-ignore-end -->
+
+There are no rules about this `addStudentData` function, and Sofia could have written this logic using any approach or code style!
+
+Sofia chose to implement `addStudentData` this way:
+
+1. The function receives a new student object, `newStudent`.
+1. She duplicates the `studentData` array into `newStudentList`, which will help React detect a change.
+1. She generates a new ID number, `nextId`.
+1. She pushes a new object into `newStudentList`. The shape of this object matches the other objects in `studentData`.
+1. The `newStudentList` now contains an object with the `newStudent` data. She updates `studentData` in state with `setStudentData`.
+
+### !callout-info
+
+## The Implied Shape of `newStudent`
+
+Consider this implementation of `addStudentData` and how Sofia accessed the name and email of the `newStudent`.
+
+<br/>
+
+This implementation reaches into `newStudent` and grabs the values of `nameData` and `emailData` specifically... not `name`, or `newName`, or anything else! We should keep in mind that this function expects `newStudent` to have the keys exactly named `nameData` and `emailData`.
+
+<br/>
+
+If we want to change the name of the expected keys, we should refactor `App`'s definition of `addStudentData`.
+
+### !end-callout
+
+Now, Sofia needs to actually use her new function. She passes it into an instance of `NewStudentForm`, through a new prop named `addStudentCallback`.
+
+<!-- prettier-ignore-start -->
+```js
+    return (
+        <main>
+            <h1>Attendance</h1>
+            <StudentList
+                students={studentData}
+                onUpdateStudent={updateStudentData}
+            ></StudentList>
+            <NewStudentForm
+                addStudentCallback={addStudentData}
+            ></NewStudentForm>
+        </main>
+    );
+```
+<!-- prettier-ignore-end -->
+
+Sofia updates the PropTypes of `NewStudentForm` to now anticipate this prop:
+
+<!-- prettier-ignore-start -->
+```js
+import PropTypes from 'prop-types';
+
+const NewStudentForm = (props) => {
+// ...
```

Comment:
I feel like this is a little too much of an elision (it loses that the prop types come outside the function)
```suggestion
    // ...
};
```